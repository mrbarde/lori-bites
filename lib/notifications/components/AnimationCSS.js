'use strict';

Object.defineProperty(exports, "__esModule", {
            value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTransitionGroup = require('react-transition-group');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CSSAnimation = function CSSAnimation(_ref) {
            var children = _ref.children,
                timeout = _ref.timeout,
                transitionName = _ref.transitionName;


            return _react2.default.createElement(
                        _reactTransitionGroup.CSSTransitionGroup,
                        {
                                    transitionEnterTimeout: timeout,
                                    transitionLeaveTimeout: timeout,
                                    transitionName: transitionName },
                        children
            );
};

exports.default = CSSAnimation;