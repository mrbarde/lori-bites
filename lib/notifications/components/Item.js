'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Notification = function (_Component) {
    _inherits(Notification, _Component);

    function Notification() {
        _classCallCheck(this, Notification);

        // initialise state
        var _this = _possibleConstructorReturn(this, (Notification.__proto__ || Object.getPrototypeOf(Notification)).call(this));

        _this.state = {
            show: true,
            timeout: null
            // bind methods
        };_this.remove = _this.remove.bind(_this);
        return _this;
    }

    _createClass(Notification, [{
        key: 'componentDidMount',
        value: function componentDidMount() {
            if (!this.props.sticky && typeof this.props.delay !== 'undefined') {
                var timeout = setTimeout(this.remove, this.props.delay);
            }
        }
    }, {
        key: 'render',
        value: function render() {
            var show = this.state.show;
            var _props = this.props,
                animation = _props.animation,
                duration = _props.duration,
                type = _props.type,
                className = _props.className,
                message = _props.message;


            type = !type ? 'info' : type;

            className = className ? " " + className : '';

            var containerStyles = Object.assign({}, styles.default.container, styles[type]['container']);
            var closeStyles = Object.assign({}, styles.default.close, styles[type]['close']);
            var messageStyles = Object.assign({}, styles.default.message, styles[type]['message']);

            return _react2.default.createElement(
                'li',
                { style: containerStyles, className: "ln_notif_" + type + className },
                _react2.default.createElement(
                    'a',
                    { href: 'javascript:;', style: closeStyles, className: 'ln_notif_close' },
                    '\xD7'
                ),
                _react2.default.createElement(
                    'p',
                    { style: messageStyles, className: 'ln_notif_message' },
                    message
                )
            );
        }
    }, {
        key: 'remove',
        value: function remove() {
            this.props.remove();
        }
    }]);

    return Notification;
}(_react.Component);

// define styles


var styles = {
    default: {
        container: {
            position: 'relative',
            float: 'left',
            width: '300px',
            marginBottom: '20px',
            padding: '30px 15px',
            backgroundColor: '#ffffff',
            border: 'solid 1px #666666'
        },
        close: {
            position: 'absolute',
            top: '10px',
            right: '15px',
            color: '#999999',
            fontSize: '14px'
        },
        message: {
            color: '#666666',
            margin: '0px 0px',
            fontSize: '13px'
        }
    },
    info: {
        container: {},
        close: {},
        message: {}
    },
    success: {
        container: {},
        close: {},
        message: {}
    },
    warning: {
        container: {},
        close: {},
        message: {}
    },
    danger: {
        container: {},
        close: {},
        message: {}
    }
};

// define component property types
Notification.propTypes = {
    message: _propTypes2.default.string.isRequired,
    type: _propTypes2.default.oneOf(['info', 'success', 'warning', 'danger']),
    className: _propTypes2.default.string,
    delay: _propTypes2.default.number.isRequired,
    sticky: _propTypes2.default.bool,
    positionStyles: _propTypes2.default.string
};

exports.default = Notification;