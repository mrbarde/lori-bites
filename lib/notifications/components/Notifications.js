'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Item = require('./Item');

var _Item2 = _interopRequireDefault(_Item);

var _redux = require('redux');

var _reactRedux = require('react-redux');

var _actions = require('../actions');

var _AnimationCSS = require('./AnimationCSS');

var _AnimationCSS2 = _interopRequireDefault(_AnimationCSS);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Notifications = function (_Component) {
    _inherits(Notifications, _Component);

    function Notifications() {
        _classCallCheck(this, Notifications);

        var _this = _possibleConstructorReturn(this, (Notifications.__proto__ || Object.getPrototypeOf(Notifications)).call(this));

        _this.state = {
            notifications: [],
            position: 'topRight',
            duration: 500,
            delay: 5000,
            animationName: 'fromLeft',
            show: true
        };
        return _this;
    }

    _createClass(Notifications, [{
        key: 'componentDidUpdate',
        value: function componentDidUpdate(prevProps) {
            var _this2 = this;

            // if new push request is made
            if (!prevProps.lnPush.isPushing && this.props.lnPush.isPushing) {
                var notifications = this.state.notifications.concat(this.props.lnPush.notification);
                // update notifications in component state
                this.setState({ notifications: notifications });
                // end push action for new notification
                this.props.pushEnd();
            }

            // if a remove request is made
            if (!prevProps.lnRemove.isRemoving && this.props.lnRemove.isRemoving) {
                // filter through notifications in state.
                var notifications = this.state.notifications.filter(function (item, index) {
                    if (item.id != _this2.props.lnRemove.notification_id) {
                        return item;
                    }
                });
                // update notifications in component state
                this.setState({ notifications: notifications });
                // end remove action for notification
                this.props.removeEnd();
            }
        }
    }, {
        key: 'componentWillMount',
        value: function componentWillMount() {
            var _props = this.props,
                position = _props.position,
                duration = _props.duration,
                delay = _props.delay,
                animationName = _props.animationName,
                notifications = _props.notifications;


            if (position) {
                this.setState({ position: position });
            }

            if (duration) {
                this.setState({ duration: duration });
            }

            if (delay) {
                this.setState({ delay: delay });
            }

            if (animationName) {
                this.setState({ animationName: animationName });
            }
        }
    }, {
        key: 'render',
        value: function render() {
            var _state = this.state,
                notifications = _state.notifications,
                position = _state.position,
                transitionName = _state.transitionName,
                timeout = _state.timeout,
                show = _state.show;

            return _react2.default.createElement(
                'ul',
                this.containerProps(),
                _react2.default.createElement(
                    _AnimationCSS2.default,
                    { transitionName: transitionName || 'loriNoti',
                        timeout: timeout || 500 },
                    notifications.map(this.renderNotification.bind(this))
                )
            );
        }
    }, {
        key: 'containerProps',
        value: function containerProps() {
            // get props
            var props = {};
            // set styles
            props.style = Object.assign({}, styles.container, styles[this.state.position], this.props.style);
            // set classname
            props.className = this.props.className ? this.props.className + " " : "";
            props.className += "lori_notifications";
            // return remaining props
            return props;
        }
    }, {
        key: 'renderNotification',
        value: function renderNotification(item, index) {
            // get default notification config
            var delay = this.state.delay;


            var notif = Object.assign({ delay: delay, type: 'info' }, item, {
                remove: this.remove.bind(this, item.id),
                positionStyles: this.state.position
            });
            return _react2.default.createElement(_Item2.default, _extends({}, notif, { key: item.id || index }));
        }
    }, {
        key: 'remove',
        value: function remove(id) {
            this.props.removeNotify(id);
        }
    }]);

    return Notifications;
}(_react.Component);

var styles = {
    container: {
        position: 'absolute',
        width: '0px',
        listStyle: 'none',
        paddingLeft: '0px',
        marginBottom: '0px',
        overflowX: 'hidden',
        overflowY: 'visible'
    },
    'topRight': {
        right: '20px',
        top: '40px'
    },
    'topLeft': {
        left: '20px',
        top: '40px'
    },
    'bottomLeft': {
        left: '20px',
        bottom: '40px'
    },
    'bottomRight': {
        right: '20px',
        bottom: '40px'
    }
};

var mapStateToProps = function mapStateToProps(_ref) {
    var lnPush = _ref.lnPush,
        lnRemove = _ref.lnRemove;
    return { lnPush: lnPush, lnRemove: lnRemove };
};

var mapDispatchToProps = function mapDispatchToProps(dispatch) {
    return (0, _redux.bindActionCreators)({ pushEnd: _actions.pushEnd, removeEnd: _actions.removeEnd, removeStart: _actions.removeStart }, dispatch);
};

exports.default = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(Notifications);