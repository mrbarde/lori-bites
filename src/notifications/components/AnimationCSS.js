import React from 'react';
import {CSSTransitionGroup} from 'react-transition-group';

const CSSAnimation = ({children, timeout, transitionName}) => {

    return(<CSSTransitionGroup 
                transitionEnterTimeout={timeout}
                transitionLeaveTimeout={timeout}
                transitionName={transitionName}>
                    {children}
            </CSSTransitionGroup>);
};

export default CSSAnimation;