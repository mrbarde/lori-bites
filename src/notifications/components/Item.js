import React, {Component} from 'react';
import propTypes from 'prop-types';

class Notification extends Component {

    constructor(){
        super();
        // initialise state
        this.state = {
            show: true,
            timeout: null
        }
        // bind methods
        this.remove = this.remove.bind(this);
    }

    componentDidMount() {
        if(!this.props.sticky && typeof this.props.delay !== 'undefined'){
            let timeout = setTimeout(this.remove, this.props.delay)
        }
    }

    render(){
        var {show} = this.state;
        var {animation, duration, type, className, message, positionStyles} = this.props;

        type = (!type) ? 'info' : type;

        className = (className) ? " "+className : '';

        let containerStyles = Object.assign({}, styles.default.container, styles[type]['container'], styles[positionStyles]);
        let closeStyles = Object.assign({}, styles.default.close, styles[type]['close']);
        let messageStyles = Object.assign({}, styles.default.message, styles[type]['message']);

        return(<li style={containerStyles} className={"ln_notif_"+type+className}>
                <a href="javascript:;" style={closeStyles} className="ln_notif_close">
                    &times;
                </a>
                <p style={messageStyles} className="ln_notif_message">{message}</p>
            </li>);
    }

    remove(){
        this.props.remove();
    }

}

// define styles
var styles = {
    default: {
        container: {
            position: 'relative',
            width: '300px',
            marginBottom: '20px',
            padding: '30px 15px',
            backgroundColor: '#ffffff',
            border: 'solid 1px #666666'
        },
        close: {
            position: 'absolute',
            top: '10px',
            right: '15px',
            color: '#999999',
            fontSize: '14px'
        },
        message: {
            color: '#666666',
            margin: '0px 0px',
            fontSize: '13px'
        }
    },
    info: {
        container: {},
        close: {},
        message: {}
    },
    success: {
        container: {},
        close: {},
        message: {}
    },
    warning: {
        container: {},
        close: {},
        message: {}
    },
    danger: {
        container: {},
        close: {},
        message: {}
    },
    topRight: {
        float: 'right'
    },
    topLeft: {
        float: 'left'
    },
    bottomRight: {
        float: 'right'
    },
    bottomLeft: {
        float: 'left'
    }
};

// define component property types
Notification.propTypes = {
    message: propTypes.string.isRequired,
    type: propTypes.oneOf(['info', 'success', 'warning', 'danger']),
    className: propTypes.string,
    delay: propTypes.number.isRequired,
    sticky: propTypes.bool,
    positionStyles: propTypes.string.isRequired
}

export default Notification;