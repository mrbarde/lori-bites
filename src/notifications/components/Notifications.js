import React, {Component} from 'react';
import Item from './Item';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {pushEnd,removeEnd} from '../actions';
import AnimationCSS from './AnimationCSS';

export class Notifications extends Component{

    constructor(){
        super();
        this.state = {
            notifications: [],
            position: 'topRight',
            duration: 500,
            delay: 5000,
            animation: 'fromLeft',
            show: true
        };
    }

    componentDidUpdate(prevProps){

        // if new push request is made
        if(!prevProps.lnPush.isPushing && this.props.lnPush.isPushing){
            var notifications = this.state.notifications
                                    .concat(this.props.lnPush.notification);
            // update notifications in component state
            this.setState({notifications});
            // end push action for new notification
            this.props.pushEnd();
        }

        // if a remove request is made
        if(!prevProps.lnRemove.isRemoving && this.props.lnRemove.isRemoving){
            // filter through notifications in state.
            var notifications = this.state.notifications.filter((item, index) => {
                if(item.id != this.props.lnRemove.notification_id){
                    return item;
                }
            });
            // update notifications in component state
            this.setState({notifications});
            // end remove action for notification
            this.props.removeEnd();
        }
    }

    componentWillMount(){
        var {position, duration, delay, animation, notifications} = this.props;

        if(position){
            this.setState({position});
        }

        if(duration){
            this.setState({duration});
        }

        if(delay){
            this.setState({delay});
        }

        if(animation){
            this.setState({animation});
        }
    }

    render(){
        var {notifications, 
            position, 
            transitionName, 
            timeout, 
            show} = this.state;
        return(
            <ul {...this.containerProps()}>
                <AnimationCSS transitionName={transitionName || 'loriNoti'}
                    timeout={timeout || 500}>
                    {notifications.map(this.renderNotification.bind(this))}
                </AnimationCSS>
            </ul>
        )
    }

    containerProps(){
        // get props
        var props = {};
        // set styles
        props.style = Object.assign({}, 
            styles.container,
            styles[this.props.position],
            this.props.style);
        // set classname
        props.className = (this.props.className) ? this.props.className+" " : "";
        props.className += "lori_notifications";
        // return remaining props
        return props;
    }

    renderNotification(item, index){
        // get default notification config
        var {delay} = this.state;

        let notif = Object.assign({delay, type: 'info'}, item, {
            remove: this.remove.bind(this, item.id)
        });
        return <Item {...notif} key={item.id || index} />;
    }

    remove(id){
        this.removeNotify(id);
    }
}

var styles = {
    container: {
        position: 'absolute',
        width: '0px',
        listStyle: 'none',
        paddingLeft: '0px',
        marginBottom: '0px',
        overflowX: 'hidden',
        overflowY: 'visible'
    },
    'topRight': {
        right: '20px',
        top: '40px'
    },
    'topLeft': {
        left: '20px',
        top: '40px'
    },
    'bottomLeft': {
        left: '20px',
        bottom: '40px'
    },
    'bottomRight': {
        right: '20px',
        bottom: '40px'
    }
};

const mapStateToProps = ({lnPush, lnRemove}) => { return {lnPush, lnRemove} };

const mapDispatchToProps = (dispatch) => { return bindActionCreators({pushEnd, removeEnd}, dispatch); }

export default connect(mapStateToProps, mapDispatchToProps)(Notifications);