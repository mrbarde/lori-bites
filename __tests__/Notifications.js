import React from 'react';
import {Notifications} from '../src/notifications/components/Notifications';
import renderer from 'react-test-renderer';

test('Notifications Container Will Render', () => {
  const component = renderer.create(
    <Notifications 
      transitionName="fade"
      position="bottomLeft"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});