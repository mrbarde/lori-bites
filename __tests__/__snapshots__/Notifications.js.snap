// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications Container Will Render 1`] = `
<ul
  className="lori_notifications"
  style={
    Object {
      "bottom": "40px",
      "left": "20px",
      "listStyle": "none",
      "marginBottom": "0px",
      "overflowX": "hidden",
      "overflowY": "visible",
      "paddingLeft": "0px",
      "position": "absolute",
      "width": "0px",
    }
  }
>
  <span />
</ul>
`;
